src/vs/workbench/browser/parts/views/customView.ts

```diff
@@ -530,136 +596,129 @@ export class CustomTreeView extends Disposable implements ITreeView {
 		collector.addRule(`.monaco-tl-contents .monaco-highlighted-label .highlight { color: unset !important; border: 1px dotted ${findMatchHighlightColorBorder}; box-sizing: border-box; }`);
 	}
 	const link = theme.getColor(textLinkForeground);
 	if (link) {
 		collector.addRule(`.tree-explorer-viewlet-tree-view > .message a { color: ${link}; }`);
 	}
-	const focustBorderColor = theme.getColor(focusBorder);
-	if (focustBorderColor) {
-		collector.addRule(`.tree-explorer-viewlet-tree-view > .message a:focus { outline: 1px solid ${focustBorderColor}; outline-offset: -1px; }`);
+	const focusBorderColor = theme.getColor(focusBorder);
+	if (focusBorderColor) {
+		collector.addRule(`.tree-explorer-viewlet-tree-view > .message a:focus { outline: 1px solid ${focusBorderColor}; outline-offset: -1px; }`);
 	}
 	const codeBackground = theme.getColor(textCodeBlockBackground);
 	if (codeBackground) {
 		collector.addRule(`.tree-explorer-viewlet-tree-view > .message code { background-color: ${codeBackground}; }`);
 	}
 });
 
-class TreeRenderer implements IRenderer {
+interface ITreeExplorerTemplateData {
+	elementDisposable: IDisposable;
+	container: HTMLElement;
+	resourceLabel: IResourceLabel;
+	icon: HTMLElement;
+	actionBar: ActionBar;
+}
 
-	private static readonly ITEM_HEIGHT = 22;
-	private static readonly TREE_TEMPLATE_ID = 'treeExplorer';
+class TreeRenderer extends Disposable implements ITreeRenderer<ITreeItem, FuzzyScore, ITreeExplorerTemplateData> {
+	static readonly ITEM_HEIGHT = 22;
+	static readonly TREE_TEMPLATE_ID = 'treeExplorer';
 
 	constructor(
 		private treeViewId: string,
 		private menus: TreeMenus,
 		private labels: ResourceLabels,
 		private actionViewItemProvider: IActionViewItemProvider,
+		private aligner: Aligner,
 		@IWorkbenchThemeService private readonly themeService: IWorkbenchThemeService,
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@ILabelService private readonly labelService: ILabelService
 	) {
+		super();
 	}
 
-	getHeight(tree: ITree, element: any): number {
-		return TreeRenderer.ITEM_HEIGHT;
-	}
-
-	getTemplateId(tree: ITree, element: any): string {
+	get templateId(): string {
 		return TreeRenderer.TREE_TEMPLATE_ID;
 	}
 
-	renderTemplate(tree: ITree, templateId: string, container: HTMLElement): ITreeExplorerTemplateData {
+	renderTemplate(container: HTMLElement): ITreeExplorerTemplateData {
 		DOM.addClass(container, 'custom-view-tree-node-item');
 
 		const icon = DOM.append(container, DOM.$('.custom-view-tree-node-item-icon'));
+
 		const resourceLabel = this.labels.create(container, { supportHighlights: true, donotSupportOcticons: true });
-		DOM.addClass(resourceLabel.element, 'custom-view-tree-node-item-resourceLabel');
-		const actionsContainer = DOM.append(resourceLabel.element, DOM.$('.actions'));
+		const actionsContainer = DOM.append(container, DOM.$('.actions'));
```

Comment:
@alexr00 One minor question as asked before, is this change intentional? Why not use the same structure as before?

Reply:
It is needed to have the correct alignment of the actions. Appending to the `resourceLabel.element` causes the action buttons to be right up against the resource label, instead of nicely aligned to the right.